### PR TITLE
button component based on bootstrap

### PR DESCRIPTION
**Pull request by DanSallau**
button

### CODE REVIEW

File: lib/src/button/button.pipes.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@Pipe({ name: "buttonClasses" })
+export class ButtonClassesPipe implements PipeTransform {
+    transform(theme: ButtonTheme, size?: ButtonSize, block?: boolean, className?: string): Array<string> {
+        const classes: Array<string> = ["btn", `btn-${theme}`];
+
+        if (size) {
+            classes.push(`btn-${size}`);
+        }
+
+        if (Boolean(block)) {
```
Comment (mario-subo) [inline lib/src/button/button.pipes.ts:13]: why not if (!!block) or even just if (block) ??

File: lib/src/button/button.module.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+import { ButtonComponent } from "./button.component";
+import { FormsModule } from "@angular/forms";
+
+import { ButtonClassesPipe } from "./button.pipes";
 
 @NgModule({
-    imports: [CommonModule],
-    declarations: [SebButtonDirective, ParseSourceExampleComponent],
-    exports: [SebButtonDirective],
+    declarations: [ButtonComponent, ButtonClassesPipe],
```
Comment (mario-subo) [inline lib/src/button/button.module.ts:9]: Why create custom pipes for each component? Can you make it generic?

File: lib/src/button/button.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Component, Input, ViewEncapsulation, EventEmitter, Output, OnInit } from "@angular/core";
+
+export type ButtonTheme = "primary" | "secondary" | "danger" | "outline-primary" | "outline-danger" | "dark" | "light" | "link";
+
+export type ButtonSize = "lg" | "md" | "sm";
+export type ButtonIconPosition = "right" | "left";
+
+export type ButtonTypes = "submit" | "button" | "reset";
```
Comment (mario-subo) [inline lib/src/button/button.component.ts:8]: I'm not sure but I think there might be a proper type for this already if you check the html button>type TS definition and see if you can use that instead.

File: lib/src/button/button.component.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        [attr.id]="id"
+        [attr.name]="name"
+        [value]="label"
+        [attr.aria-disabled]="disabled"
+        [attr.aria-pressed]="active"
+        [class.active]="active"
+    />
+
+    <a
+        href="#"
```
Comment (mario-subo) [inline lib/src/button/button.component.html:17]: I would like an option to pass the href if I am using this component as an anchor tag

File: lib/src/button/button.pipes.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            classes.push("btn-block");
+        }
+
+        if (className) {
+            classes.push(className);
+        }
```
Comment (yousifalraheem) [inline lib/src/button/button.pipes.ts:19]:
```suggestion
        !!size && classes.push(`btn-${size}`);
        !!block && classes.push("btn-block");
        !!className && classes.push(className);
```

File: lib/src/button/button.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+import { ButtonTheme, ButtonSize, ButtonTags, ButtonComponent } from "./button.component";
+import { By } from "@angular/platform-browser";
+import { Component } from "@angular/core";
+import { ButtonClassesPipe } from "./button.pipes";
+import { CommonModule } from "@angular/common";
+
+type ButtonTestItem<T, K> = { value: T; expected: K };
+
+@Component({
+    selector: "test-sebng-chip",
```
Comment (yousifalraheem) [inline lib/src/button/button.component.spec.ts:12]:
```suggestion
    selector: "test-sebng-button",
```

**Comment by yousifalraheem** [comment]
Can you change the example options usage of dropdown to just checkboxes?

**Comment by mario-subo** [comment]
@DanSallau did you see #186 ? Does your PR address this? 

Also another one related to button ->  #82

**Pull request merged**